pulp_2to3_migrate/app/json_schema.py

```diff
@@ -0,0 +1,89 @@
+                "properties": {
+                    "type": {
+                        "type": "string"
+                    },
+                    "protection": {
+                        "type": "boolean"
+                    },
+                    "repositories": {
+                        "type": "array",
+                        "items": {
+                            "type": "object",
+                            "properties": {
+                                "name": {
+                                    "type": "string"
+                                },
+                                "pulp2_repository_id": {
+                                    "type": "string"
+                                },
+                                "repository_versions": {
+                                    "type": "array",
+                                    "minItems": 1,
+                                    "items": {
+                                        "type": "object",
+                                        "properties": {
+                                            "pulp2_repository_id": {
+                                                "type": "string"
+                                             },
+                                             "distributor_ids": {
+                                                 "type": "array"
+                                             },
+                                             "protection": {
+                                                 "type": "array"
+                                              }
+                                        },
+                                        "required": ["pulp2_repository_id"],
+                                        "additionalProperties": false,
+                                        "dependencies": {
+                                            "protection": [ "distributor_ids"]
+                                        }
+                                    }
+                                }
+                            },
+                            "required": ["name", "repository_versions"],
+                            "additionalProperties": false,
+			    "if": {
+    			        "properties": {
+                                    "repository_versions": {
+                                        "type": "array",
+                                        "minItems": 2
+                                    }
+                                }
+                            },
+			    "then": {
+    		                "dependencies": {
+                                    "repository_versions": ["pulp2_repository_id"]
+                                }
+                             }
+                        }
+                    }
+                },
+                "required": ["type"],
+                "additionalProperties": false,
+                "dependencies": {
+                    "protection": {
```

Comment:
I'm not sure I understand this dependency.
For global `protection` we require at least one item in the repositories list?
If so, I think just `"required": ["repositories"]` is enough - we treat empty list as migrate all, so we can migrate global protection for all the repos as well.
Let me know if I missed anything.

Reply:
i made protection and repositories interdependent